src/v2/Apps/Article/Components/ArticleHTML.tsx

```diff
@@ -1,13 +1,43 @@
-import { Box, BoxProps, THEME_V3 as THEME } from "@artsy/palette"
+import { Box, BoxProps, THEME_V3 as THEME, useDidMount } from "@artsy/palette"
 import { themeGet } from "@styled-system/theme-get"
 import { FC } from "react"
 import styled from "styled-components"
+import reactHtmlParser from "@artsy/react-html-parser"
+import { ArticleTooltip, isSupportedArticleTooltip } from "./ArticleTooltip"
 
 interface ArticleHTMLProps extends BoxProps {
   children: string
 }
 
 export const ArticleHTML: FC<ArticleHTMLProps> = ({ children, ...rest }) => {
+  const isMounted = useDidMount()
+
+  // Looks for links and if they are internal and a supported entity type,
+  // inserts the relevant tooltip.
+  const transform = (node: Element) => {
+    if (node.tagName !== "A") return
+
+    const { href } = node as HTMLAnchorElement
+
+    const uri = new URL(href)
+
+    if (!uri.hostname.includes("artsy.net")) return
+
+    const [_, entity, id] = uri.pathname.split("/")
+
+    if (!isSupportedArticleTooltip(entity)) return
+
+    return (
+      <ArticleTooltip entity={entity} id={id} href={href}>
+        {node.textContent}
+      </ArticleTooltip>
+    )
+  }
+
+  if (isMounted) {
+    return <Container>{reactHtmlParser(children, { transform })}</Container>
+  }
+
```

Comment:
Here we use our fork of `reactHtmlParser` — which "replaces the `htmlparser2` dependency with the `DOMParser` API provided by modern browsers" — this is great because the package's file size is _dramatically_ reduced and we only need to run this on the client-side anyway.

So on the server side: this returns the normal HTML, on the client-side it returns an identical looking body but with the tooltips inserted.